pkg/mtu/detect_linux.go

```diff
@@ -75,3 +75,33 @@ func autoDetect() (int, error) {
 
 	return EthernetMTU, nil
 }
+
+// getMTUFromIf finds the interface that holds the ip and returns its mtu
+func getMTUFromIf(ip net.IP) (int, error) {
+	ifaces, err := net.Interfaces()
+	if err != nil {
+		return 0, fmt.Errorf("Unable to list interfaces: %s", err)
+	}
+
+	for _, iface := range ifaces {
+		addrs, err := iface.Addrs()
+		if err != nil {
+			return 0, fmt.Errorf("Unable to list the addresses: %s", err)
```

Comment:
As far as I see, the error can be triggered because:

1 - iface is nil
2 - The syscalls return an error:
       2.1 - syscall.NetlinkRIB returns an error
       2.2 - syscall.ParseNetlinkMessage returns an error

If any of those errors happen, there will be no slice of addresses and therefore we cannot continue. I am thinking now that if there is an error here, we should default to calculate the mtu as it was originally done, to at least have a value

Reply:
@joestringer ok, I have just realized that I already did that, if there is an error then we will default mtu = EthernetMTU which is 1500. Would substituting the Errors with Warnings solve your concerns? 

107                 if nodeInternalIP != nil {
108                         mtu, err = getMTUFromIf(nodeInternalIP)
109                 } else {
110                         mtu, err = autoDetect()
111                 }
112                 if err != nil {
113                         log.WithError(err).Warning("Unable to automatically detect MTU")
114                         mtu = EthernetMTU
115                 }

Reply:
My concern was specifically if we have a device list like:

- random dev
- random dev that fails in some way (eg fail to dump addresses)
- device with the IP we're looking for

Then even if we would otherwise be able to fetch the addresses from the right device, the failure from a previous device may cause the autodetection to fail overall.

It's probably a bit far-fetched that we hit a case like this, but I wanted to at least ask the queston to see whether it's worth taking a `log; continue` approach in this detection logic.

Reply:
Ok, I understand what you mean now. You are right, it would fail and we don't want that. Let me change the code following your suggestion